ansible/roles/test/files/ptftests/advanced-reboot.py

```diff
@@ -664,124 +690,127 @@ def runTest(self):
         self.log("Check that device is alive and pinging")
         self.assertTrue(self.check_alive(), 'DUT is not stable')
 
-        self.log("Schedule to reboot the remote switch in %s sec" % self.reboot_delay)
-        thr.start()
-
-        self.log("Wait until VLAN and CPU port down")
-        self.timeout(self.task_timeout, "DUT hasn't shutdown in %d seconds" % self.task_timeout)
-        self.wait_until_vlan_cpu_port_down()
-        self.cancel_timeout()
+        try:
+            self.log("Schedule to reboot the remote switch in %s sec" % self.reboot_delay)
```

Comment:
It cover many cases. One is:
1. 'warm-reboot' may fail, and it will throw a KeyboardInterrupt exception
2. try block catch it and come into final block

Another one is interrupt a long running (or infinite loop) test, we still get some meaningful log output. 

---
In reply to: [240769373](https://github.com/Azure/sonic-mgmt/pull/758#discussion_r240769373) [](ancestors = 240769373)